### PR TITLE
fix formal shadowing issue

### PR DESCRIPTION
**Pull request by stylewarning**
This fixes the issue described in https://github.com/rigetti/qvm/issues/236

### CODE REVIEW

**Comment by stylewarning** [comment]
another global special never hurts, right?

File: src/parser.lisp [hunk 1]
```diff
@@ -647,22 +647,42 @@ If ENSURE-VALID is T (default), then a memory reference such as 'foo[0]' will re
...
                            *parse-context*))
-       (if names-memory-region-p
-           (mref (token-payload tok) 0)
-           (formal (token-payload tok)))))
+       (cond
+         ((and *formal-arguments-allowed*
+               shadowing-formal)
+          shadowing-formal)
+         (names-memory-region-p (mref (token-payload tok) 0))
+         (t (formal (token-payload tok))))))
```
Comment (appleby) [inline src/parser.lisp]: In what situation do you wind up in the default clause here?

Comment (stylewarning) [reply src/parser.lisp]: Hopefully never. Maybe the least I can do is warn.

**Comment by appleby** [comment]
> illusive

illusive

**Pull request merged**